suite/tests/common/decode.c

```diff
@@ -366,6 +376,82 @@ GLOBAL_LABEL(FUNCNAME:)
+        RAW(c5) RAW(f9) RAW(44) RAW(da)                 /* knotb  %k2,%k3 */
+        RAW(c4) RAW(e1) RAW(f8) RAW(44) RAW(ec)         /* knotq  %k4,%k5 */
+        RAW(c4) RAW(e1) RAW(f9) RAW(44) RAW(fe)         /* knotd  %k6,%k7 */
+        RAW(c5) RAW(f4) RAW(45) RAW(d0)                 /* korw   %k0,%k1,%k2 */
+        RAW(c5) RAW(dd) RAW(45) RAW(eb)                 /* korb   %k3,%k4,%k5 */
+        RAW(c4) RAW(e1) RAW(c4) RAW(45) RAW(c6)         /* korq   %k6,%k7,%k0 */
+        RAW(c4) RAW(e1) RAW(ed) RAW(45) RAW(d9)         /* kord   %k1,%k2,%k3 */
+        RAW(c5) RAW(f4) RAW(46) RAW(d0)                 /* kxnorw %k0,%k1,%k2 */
+        RAW(c5) RAW(dd) RAW(46) RAW(eb)                 /* kxnorb %k3,%k4,%k5 */
+        RAW(c4) RAW(e1) RAW(c4) RAW(46) RAW(c6)         /* kxnorq %k6,%k7,%k0 */
+        RAW(c4) RAW(e1) RAW(ed) RAW(46) RAW(d9)         /* kxnord %k1,%k2,%k3 */
+        RAW(c5) RAW(f4) RAW(47) RAW(d0)                 /* kxorw  %k0,%k1,%k2 */
+        RAW(c5) RAW(dd) RAW(47) RAW(eb)                 /* kxorb  %k3,%k4,%k5 */
+        RAW(c4) RAW(e1) RAW(c4) RAW(47) RAW(c6)         /* kxorq  %k6,%k7,%k0 */
+        RAW(c4) RAW(e1) RAW(ed) RAW(47) RAW(d9)         /* kxord  %k1,%k2,%k3 */
+        RAW(c5) RAW(f4) RAW(4a) RAW(d0)                 /* kaddw  %k0,%k1,%k2 */
+        RAW(c5) RAW(dd) RAW(4a) RAW(eb)                 /* kaddb  %k3,%k4,%k5 */
+        RAW(c4) RAW(e1) RAW(c4) RAW(4a) RAW(c6)         /* kaddq  %k6,%k7,%k0 */
+        RAW(c4) RAW(e1) RAW(ed) RAW(4a) RAW(d9)         /* kaddd  %k1,%k2,%k3 */
+        RAW(c5) RAW(f8) RAW(98) RAW(c8)                 /* kortestw %k0,%k1 */
+        RAW(c5) RAW(f9) RAW(98) RAW(da)                 /* kortestb %k2,%k3 */
+        RAW(c4) RAW(e1) RAW(f8) RAW(98) RAW(ec)         /* kortestq %k4,%k5 */
+        RAW(c4) RAW(e1) RAW(f9) RAW(98) RAW(fe)         /* kortestd %k6,%k7 */
+        RAW(c5) RAW(f8) RAW(99) RAW(c8)                 /* ktestw %k0,%k1 */
+        RAW(c5) RAW(f9) RAW(99) RAW(da)                 /* ktestb %k2,%k3 */
+        RAW(c4) RAW(e1) RAW(f8) RAW(99) RAW(ec)         /* ktestq %k4,%k5 */
+        RAW(c4) RAW(e1) RAW(f9) RAW(99) RAW(fe)         /* ktestd %k6,%k7 */
+        RAW(c4) RAW(e3) RAW(f9) RAW(32) RAW(c8) RAW(ff) /* kshiftlw $0xff,%k0,%k1 */
+        RAW(c4) RAW(e3) RAW(79) RAW(32) RAW(da) RAW(7b) /* kshiftlb $0x7b,%k2,%k3 */
+        RAW(c4) RAW(e3) RAW(f9) RAW(33) RAW(ec) RAW(07) /* kshiftlq $0x7,%k4,%k5 */
+        RAW(c4) RAW(e3) RAW(79) RAW(33) RAW(fe) RAW(63) /* kshiftld $0x63,%k6,%k7 */
+        RAW(c4) RAW(e3) RAW(f9) RAW(30) RAW(c8) RAW(df) /* kshiftrw $0xdf,%k0,%k1 */
+        RAW(c4) RAW(e3) RAW(79) RAW(30) RAW(da) RAW(65) /* kshiftrb $0x65,%k2,%k3 */
+        RAW(c4) RAW(e3) RAW(f9) RAW(31) RAW(ec) RAW(05) /* kshiftrq $0x5,%k4,%k5 */
+        RAW(c4) RAW(e3) RAW(79) RAW(31) RAW(fe) RAW(2f) /* kshiftrd $0x2f,%k6,%k7 */
+        RAW(c4) RAW(e1) RAW(79) RAW(31) RAW(fe) RAW(2f) /* bad */
```

Comment:
How does this test tell us whether it properly decoded any of the above?  It seems like it should have two separate functions, one with nothing bad which does not fault (has to check for avx512 support on processor first) and one with sthg bad.

Reply:
Note that I put this in as a bonus, AVX(2) didn't have anything. I took this from the sse3 and 3dnow test, they seem to have the same issue. But you're right, the "bad" instruction doesn't make sense. I've removed the "bad" instance from the test.